### PR TITLE
Bump System.Security.Cryptography.Xml version to address CVE-2023-29331

### PR DESCRIPTION
**Pull request by RussKie**
https://dev.azure.com/dnceng-public/public/_componentGovernance/201695/alert/7938738?typeId=21448446

### CODE REVIEW

File: eng/tools/RepoTasks/RepoTasks.csproj [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" />
     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreVersion)" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" />
-    <!-- Manually updated version from 6.0.0 to address CVE-2021-43877 -->
+    <!-- Manually updated version from 6.0.0 to address CVE-2023-29331 -->
```
Comment (BrennanConroy) [inline eng/tools/RepoTasks/RepoTasks.csproj:29]: 8.0?

Comment (RussKie) [reply eng/tools/RepoTasks/RepoTasks.csproj:29]: Not sure how to interpret your comment, @BrennanConroy, sorry...

Comment (BrennanConroy) [reply eng/tools/RepoTasks/RepoTasks.csproj:29]: Update the comment to say 8.0, not 6.0

Comment (danmoseley) [reply eng/tools/RepoTasks/RepoTasks.csproj:29]: It says "from 6.0" not to?

**Comment by danmoseley** [comment]
I'm not at my desk to check. Is this a transitive dependency that we're only directly depending on in order to pull it forward for the CVE? If so updating it is only a nice to have, if I understand correctly.

**Comment by wtgodbe** [comment]
> I'm not at my desk to check. Is this a transitive dependency that we're only directly depending on in order to pull it forward for the CVE? If so updating it is only a nice to have, if I understand correctly.

Even if it's transitive, RepoTasks is a tool, not a library, so we have to fix it either way

**Comment by wtgodbe** [comment]
> /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Formats.Asn1.8.0.0
/__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Security.Cryptography.Pkcs.8.0.0
/__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Security.Cryptography.Xml.8.0.0

@MichaelSimons @mthalman was System.Security.Crytpo.Xml 6.0.0 (and its dependencies) already special-cased for source-build? They're not in our repo baseline

**Comment by MichaelSimons** [comment]
> > /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Formats.Asn1.8.0.0
> > /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Security.Cryptography.Pkcs.8.0.0
> > /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Security.Cryptography.Xml.8.0.0
> 
> @MichaelSimons @mthalman was System.Security.Crytpo.Xml 6.0.0 (and its dependencies) already special-cased for source-build? They're not in our repo baseline

System.Security.Crytpo.Xml 6.0.1 was being [supplied by SBRP](https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/system.security.cryptography.xml/6.0.1).  With this change, the 8.0 version and its dependencies would need to be added. ([instructions](https://github.com/dotnet/source-build-reference-packages?tab=readme-ov-file#adding-new-packages)).

**Comment by RussKie** [comment]
> > > /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Formats.Asn1.8.0.0
> > > /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Security.Cryptography.Pkcs.8.0.0
> > > /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24212.4/tools/SourceBuild/AfterSourceBuild.proj(81,5): error : System.Security.Cryptography.Xml.8.0.0
> > 
> > 
> > @MichaelSimons @mthalman was System.Security.Crytpo.Xml 6.0.0 (and its dependencies) already special-cased for source-build? They're not in our repo baseline
> 
> System.Security.Crytpo.Xml 6.0.1 was being [supplied by SBRP](https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/system.security.cryptography.xml/6.0.1?rgh-link-date=2024-04-23T17%3A17%3A00Z). With this change, the 8.0 version and its dependencies would need to be added. ([instructions](https://github.com/dotnet/source-build-reference-packages?tab=readme-ov-file#adding-new-packages)).

Thank you, I'll look into this.

**Comment by danmoseley** [comment]
> Even if it's transitive, RepoTasks is a tool, not a library, so we have to fix it either way

Gotcha, agreed.

**Review by javiercn — APPROVED** [review]

**Pull request merged**